### PR TITLE
#750: Supporting new recommendation headers

### PR DESCRIPTION
**Pull request by hani-ibrahim**
fixing #750

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasSDK/AtlasSDK/API/Endpoints/GetArticleRecommendationsEndpoint.swift [hunk 1]
```diff
@@ -4,6 +4,30 @@
...
 
+/**
+ Header Configuration needed to the get better recommendations
+ 
+ - Note: Please contact us to figure out the best recommendations properties for you
+ */
+public struct RecommendationConfig {
+
+    /// Recommendation type like similar product or product from the same brand
+    public let type: String
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/API/Endpoints/GetArticleRecommendationsEndpoint.swift]: Man, use `enum` here!

File: AtlasSDK/AtlasSDK/API/Endpoints/GetArticleRecommendationsEndpoint.swift [hunk 2]
```diff
@@ -4,6 +4,30 @@
...
+
+    /// Recommendation location as where it is disaplyed in the channel
+    public let location: String
+
+    /// Recommendation channel used to display the recommendation like web or mobile
+    public let channel: String
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/API/Endpoints/GetArticleRecommendationsEndpoint.swift]: `enum`

File: AtlasSDK/AtlasSDK/API/Endpoints/GetArticleRecommendationsEndpoint.swift [hunk 3]
```diff
@@ -4,6 +4,30 @@
...
+ 
+ - Note: Please contact us to figure out the best recommendations properties for you
+ */
+public struct RecommendationConfig {
+
+    /// Recommendation type like similar product or product from the same brand
+    public let type: String
+
+    /// Recommendation location as where it is disaplyed in the channel
+    public let location: String
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/API/Endpoints/GetArticleRecommendationsEndpoint.swift]: Maybe `enum` as well?

File: AtlasUI/AtlasUI/CheckoutSummaryRecommendationStackView.swift [hunk 4]
```diff
@@ -65,7 +65,8 @@ extension CheckoutSummaryRecommendationStackView: UIDataBuilder {
     typealias T = Article
 
     func configure(viewModel: T) {
-        AtlasAPI.withLoader.recommendations(for: viewModel.id, onView: loaderContrainer) { [weak self] result in
+        let recommendationConfig = RecommendationConfig(type: "cross_sell", location: "atlas_ios_order_confirmation", channel: "app")
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryRecommendationStackView.swift]: Would be good to use it as `RecommendationConfig.crossSell(location: .orderConfirmation)` or similar.

**Comment by bonkey** [comment]
👍

**Pull request merged**